bpf/lib/nodeport_egress.h

```diff
@@ -100,13 +100,19 @@ int tail_handle_snat_fwd_ipv6(struct __ctx_buff *ctx)
 	};
 	enum trace_point obs_point;
 	union v6addr saddr = {};
+	__u8 encryption_bit = 0;
 	int ret;
 	__s8 ext_err = 0;
 
 #ifdef IS_BPF_OVERLAY
 	obs_point = TRACE_TO_OVERLAY;
 #else
 	obs_point = TRACE_TO_NETWORK;
+
+#ifdef ENABLE_WIREGUARD
+	if ((ctx->mark & MARK_MAGIC_WG_ENCRYPTED) == MARK_MAGIC_WG_ENCRYPTED)
+		encryption_bit = TRACE_REASON_ENCRYPTED;
+#endif /* ENABLE_WIREGUARD */
 #endif
```

Comment:
Before we start adding this check to all the deep-down NAT flows, I think it makes sense to reconsider whether our WG traffic should even go down that path. Or be excluded [here](https://github.com/cilium/cilium/blob/0e4bc5e39ba2927f8c829c7f9f0626cd5b0f2e05/bpf/bpf_host.c#L1617).

We clearly don't care about the RevDNAT case for service replies. So it comes down to understanding the SNAT case a bit better.